### DISCUSSION TITLE
Change of activity type

### DISCUSSION THREAD

**Discussion by zell-mbc** [OpenTracksApp/OpenTracks#927] [Q&A] [answered]
First of all, thanks a lot for this app! 

Maybe I have completely missed the expected way to change this setting, but is settings/advanced really the only way to change the activity type?
I change my activity types frequently and keep on forgetting where this setting is hiding, probably getting old :-)
If it is, how about allowing to change and start the activity via a long press on the start button?
-> short press starts default activity
-> long press opens select-activity-type menu and starts selected activity

**Comment by dennisguse** [comment, marked as answer]
In until v3.19.X it is also possible while recording by clicking the activity icon on the statistics UI.
As this was rewritten in v3.20.X, this is not possible anymore.

And while recording: you can also use the "Edit" option in the overflow menu.

**Comment by zell-mbc** [comment]
Thanks @dennisguse, I didn't notice the edit option yet. Probably because I am "active" after starting the activity and don't check the screen until I stop the activity :-)

**Comment by WPettersson** [comment]
Just to jump onto the same conversation, I'm also looking for a workflow for this. I mostly cycle, but do run fairly often, so my default activity is cycling but I really do like switching activities so that my runs are reported in mins/km pace rather than a km/h speed. However, running is also tricky to find in the long list of activities, so rather than "Start activity, spend 10 seconds finding the right activity, then start actually running" I've now moved to "Start activity, immediately pause activity, search for running and select it, resume activity, start running". It works for now, but one slightly improved option would be to make the long press on the start button start and immediately pause an activity, to let the user set such preferences without starting the timer or tracking.

**Reply by dennisguse** [reply]
PR welcome ;)
We needs this for other things as 
* https://github.com/OpenTracksApp/OpenTracks/issues/1259
* https://github.com/OpenTracksApp/OpenTracks/issues/240
